benchmark/benchmark.py

```diff
@@ -134,6 +134,7 @@ def iterate(seq: Any) -> None:
         ("${a:b,c,d}", ValueKind.INTERPOLATION),
         ("${${b}}", ValueKind.INTERPOLATION),
         ("${a:${b}}", ValueKind.INTERPOLATION),
+        ("${foooooooooooooooooooo}_${bar}", ValueKind.INTERPOLATION),
```

Comment:
1. Add this case (or a slightly easier one) to benchmark/benchmark.py (test_get_value_kind) and show the results before and after.

2. Add an even bigger one to the benchmark, if this ever happens again I want us to notice (no need to show the 'before' result for it).

Reply:
So, this was supposed to be a good benchmark (this is actually in benchmark.py), but apparently I made a mistake because it wasn't that bad before after all (I think it's because at some point I was testing on a simplified version of the grammar where this example was slow, but it turns out it's faster with the full grammar).

I added the example from #622 instead in 3c7a9db.

Here's the before / after comparison (NOW = after):

```
Name (time in us)                                                                                                              Min                        Max                       Mean                  StdDev                     Median                     IQR            Outliers           OPS            Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_get_value_kind[${${b}}-ValueKind.INTERPOLATION-False] (0002_ec69e33)                                                   4.9730 (1.25)            180.2540 (6.64)              5.8947 (1.37)           3.0550 (3.04)              5.3800 (1.28)           0.8610 (14.59)   1018;1218  169,644.3086 (0.73)      57871           1
test_get_value_kind[${${b}}-ValueKind.INTERPOLATION-False] (NOW)                                                            4.0730 (1.02)            119.3380 (4.39)              4.3328 (1.01)           1.5545 (1.55)              4.2400 (1.01)           0.0640 (1.08)     376;2837  230,798.2216 (0.99)      88842           1
test_get_value_kind[${${b}}-ValueKind.INTERPOLATION-True] (0002_ec69e33)                                                  371.6350 (93.26)           568.0790 (20.92)           394.3330 (91.91)         35.8492 (35.68)           379.1090 (90.35)         22.9475 (388.94)      12;12    2,535.9277 (0.01)        125           1
test_get_value_kind[${${b}}-ValueKind.INTERPOLATION-True] (NOW)                                                           251.5990 (63.14)         2,458.9400 (90.55)           295.1087 (68.78)        179.9158 (179.06)          261.0190 (62.21)         38.7710 (657.14)       1;14    3,388.5820 (0.01)        154           1
test_get_value_kind[${a:${b}}-ValueKind.INTERPOLATION-False] (0002_ec69e33)                                                 4.3850 (1.10)             61.1450 (2.25)              6.3270 (1.47)           2.0309 (2.02)              5.6830 (1.35)           1.7040 (28.88)   4208;3017  158,053.2254 (0.68)      61512           1
test_get_value_kind[${a:${b}}-ValueKind.INTERPOLATION-False] (NOW)                                                          3.9950 (1.00)            102.4350 (3.77)              4.6010 (1.07)           1.6141 (1.61)              4.2360 (1.01)           0.1070 (1.81)   6280;18295  217,344.8259 (0.93)      88316           1
test_get_value_kind[${a:${b}}-ValueKind.INTERPOLATION-True] (0002_ec69e33)                                                404.7260 (101.56)       13,541.5380 (498.68)          518.1152 (120.76)       700.5439 (697.21)          423.5355 (100.94)        62.5520 (>1000.0)      2;50    1,930.0729 (0.01)        362           1
test_get_value_kind[${a:${b}}-ValueKind.INTERPOLATION-True] (NOW)                                                         332.0100 (83.31)           583.6240 (21.49)           373.5459 (87.06)         42.4297 (42.23)           376.8215 (89.80)         47.7620 (809.53)        9;5    2,677.0473 (0.01)         78           1
test_get_value_kind[${a:b,c,d}-ValueKind.INTERPOLATION-False] (0002_ec69e33)                                                5.8350 (1.46)            104.5330 (3.85)              6.4658 (1.51)           1.2373 (1.23)              6.3600 (1.52)           0.2470 (4.19)     488;1140  154,659.6274 (0.66)      40295           1
test_get_value_kind[${a:b,c,d}-ValueKind.INTERPOLATION-False] (NOW)                                                         4.0660 (1.02)             92.5620 (3.41)              4.3133 (1.01)           1.4871 (1.48)              4.2200 (1.01)           0.0630 (1.07)     293;2521  231,841.2422 (0.99)      70265           1
test_get_value_kind[${a:b,c,d}-ValueKind.INTERPOLATION-True] (0002_ec69e33)                                                 8.3650 (2.10)             83.0630 (3.06)              9.1106 (2.12)           1.3598 (1.35)              8.9690 (2.14)           0.3080 (5.22)     661;1215  109,762.5355 (0.47)      25470           1
test_get_value_kind[${a:b,c,d}-ValueKind.INTERPOLATION-True] (NOW)                                                          5.6800 (1.43)             62.8900 (2.32)              6.1945 (1.44)           1.4846 (1.48)              5.9300 (1.41)           0.1770 (3.00)    518;10557  161,433.2178 (0.69)      51763           1
test_get_value_kind[${a}-ValueKind.INTERPOLATION-False] (0002_ec69e33)                                                      4.3590 (1.09)            100.2410 (3.69)              5.4913 (1.28)           2.3274 (2.32)              5.1480 (1.23)           1.7580 (29.80)    1019;611  182,104.7479 (0.78)      57281           1
test_get_value_kind[${a}-ValueKind.INTERPOLATION-False] (NOW)                                                               4.0800 (1.02)            126.6430 (4.66)              4.2905 (1.0)            1.3466 (1.34)              4.2220 (1.01)           0.0590 (1.0)      275;2130  233,075.6835 (1.0)       85617           1
test_get_value_kind[${a}-ValueKind.INTERPOLATION-True] (0002_ec69e33)                                                       5.2190 (1.31)            126.2610 (4.65)              6.5921 (1.54)           4.2725 (4.25)              5.7980 (1.38)           0.3860 (6.54)    2656;3969  151,697.1096 (0.65)      55854           1
test_get_value_kind[${a}-ValueKind.INTERPOLATION-True] (NOW)                                                                5.1610 (1.30)            132.6410 (4.88)              7.4769 (1.74)           1.5178 (1.51)              7.7030 (1.84)           0.3320 (5.63)    3453;8029  133,745.7508 (0.57)      36802           1
test_get_value_kind[${long_string1xxx}_${long_string2xxx:${key}}-ValueKind.INTERPOLATION-False] (0002_ec69e33)              4.1590 (1.04)             54.4840 (2.01)              4.9441 (1.15)           1.3077 (1.30)              4.3850 (1.05)           0.9920 (16.81)    4097;220  202,260.8630 (0.87)      53574           1
test_get_value_kind[${long_string1xxx}_${long_string2xxx:${key}}-ValueKind.INTERPOLATION-False] (NOW)                       4.3650 (1.10)             71.9000 (2.65)              6.2213 (1.45)           1.0048 (1.0)               6.2430 (1.49)           0.1370 (2.32)   6706;10677  160,738.3555 (0.69)      61985           1
test_get_value_kind[${long_string1xxx}_${long_string2xxx:${key}}-ValueKind.INTERPOLATION-True] (0002_ec69e33)      20,620,578.1350 (>1000.0)  22,087,403.4510 (>1000.0)  21,413,115.1008 (>1000.0)  568,256.5710 (>1000.0)  21,351,155.9770 (>1000.0)  827,582.5837 (>1000.0)       2;0        0.0467 (0.00)          5           1
test_get_value_kind[${long_string1xxx}_${long_string2xxx:${key}}-ValueKind.INTERPOLATION-True] (NOW)                      784.2530 (196.80)        1,867.2150 (68.76)           869.5613 (202.67)       129.0765 (128.46)          845.4060 (201.48)        37.6490 (638.12)       8;31    1,150.0051 (0.00)        251           1
test_get_value_kind[simple-ValueKind.VALUE-False] (0002_ec69e33)                                                            4.0010 (1.00)             92.3330 (3.40)              4.5073 (1.05)           2.3988 (2.39)              4.2090 (1.00)           0.0860 (1.46)    1348;3999  221,862.6213 (0.95)      81921           1
test_get_value_kind[simple-ValueKind.VALUE-False] (NOW)                                                                     4.0760 (1.02)             69.4070 (2.56)              5.7898 (1.35)           1.3164 (1.31)              6.2000 (1.48)           1.2120 (20.54)   18891;290  172,718.3233 (0.74)      88992           1
test_get_value_kind[simple-ValueKind.VALUE-True] (0002_ec69e33)                                                             3.9850 (1.0)              27.1550 (1.0)               4.5250 (1.05)           2.1217 (2.11)              4.2110 (1.00)           0.1210 (2.05)      167;316  220,996.5590 (0.95)       8203           1
test_get_value_kind[simple-ValueKind.VALUE-True] (NOW)                                                                      4.0680 (1.02)             78.3300 (2.88)              4.3411 (1.01)           1.7225 (1.71)              4.1960 (1.0)            0.0620 (1.05)     103;1675  230,354.7968 (0.99)      17871           1
```